script.js

```diff
@@ -0,0 +1,219 @@
+const notionJsonTextInput = document.getElementById("notionMimeType");
+
+function printOriginal(clipboardData) {
+  const clipboardContentElement = document.getElementById("original");
+  clipboardContentElement.innerHTML = "";
+  for (const type of clipboardData.types) {
+    const clipData = clipboardData.getData(type);
+    let h3 = document.createElement("h3");
+    h3.textContent = `Content type: ${type}`;
+    clipboardContentElement.appendChild(h3);
+    const resultElement = document.createElement("pre");
+    resultElement.style = "white-space: pre-wrap; word-wrap: break-word;";
+    resultElement.textContent = clipData;
+    clipboardContentElement.appendChild(resultElement);
+  }
+}
+
+function parseGithubHtml(htmlString) {
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(htmlString, "text/html");
+
+  const rows = doc.querySelectorAll(".blob-code");
+  const codeLines = [];
+
+  let lastFlag = "none";
+
+  rows.forEach((row) => {
+    if (row.classList.contains("blob-code")) {
+      // Check if this line represents addition, deletion, or none
+      if (row.classList.contains("blob-code-addition")) {
+        lastFlag = "added";
+      } else if (row.classList.contains("blob-code-deletion")) {
+        lastFlag = "removed";
+      } else {
+        lastFlag = "none";
+      }
+      // Extract code content and create object
+      const line = row.innerText;
+      codeLines.push({
+        line,
+        flag: lastFlag,
+      });
+    }
+  });
+
+  return codeLines;
+}
+
+function extractFileNameFromHtml(htmlString) {
+    const filenamePattern = /<button data-path="([^"]+)"/;
+    const filenameMatch = htmlString.match(filenamePattern);
+    if (filenameMatch) {
+        return filenameMatch[1].toLowerCase();
```

Comment:
leave the responsability of lowercase to the mapping function (it's already doing it)
```suggestion
        return filenameMatch[1];
```